src/contract.rs

```diff
@@ -31,15 +31,35 @@ impl Contract {
         crypto::ephemeral_merkle_root(root)
     }
 
+    /// Calculate the root of the initial storage slots for this contract
+    pub fn initial_storage_root(storage_slots: &[StorageSlot]) -> Bytes32 {
```

Comment:
Hmmm. Should this be in `fuel-tx` instead? It's something that is entirely independent from the VM. I'm not against keeping it here for now, since there are a lot of things in the VM already that don't really belong and we need to do a big sweep one day.

Also, maybe add a TODO here to switch to SMT when ready?

Reply:
That would make sense, however the contract id calculation logic currently lives in the VM so I figured that would be the most logical place for now.